GitBasic/Controls/ConsoleControl.xaml.cs

```diff
@@ -176,5 +177,28 @@ private void EnterText(string text, int selectionIndex)
         private const string CD = "cd";
         private const string GIT_STATUS = "git status";
         private const string COMMIT_ALL = "git commit -a -m \"\"";
+
+
+        private string _defaultDirectory => Environment.GetFolderPath(Environment.SpecialFolder.Desktop);
+
+        #region Dependency Properties
+
+        public string WorkingDirectory
+        {
+            get { return (string)GetValue(WorkingDirectoryProperty); }
+            set { SetValue(WorkingDirectoryProperty, value); }
+        }        
+        public static readonly DependencyProperty WorkingDirectoryProperty =
+            DependencyProperty.Register("WorkingDirectory", typeof(string), typeof(ConsoleControl), new PropertyMetadata(string.Empty, OnWorkingDirectoryChanged));
+
+        private static void OnWorkingDirectoryChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
```

Comment:
This is a callback that gets raised anytime the value of the WorkingDirectory dependency property gets changed.
In this case, I am using it to persist the working directory.